File: week07/frequency.py
#!/usr/bin/env python

import fileinput, re

count = {}

for linei n fileinput.input():
    for word in re.findall(r'\w', line.lower()):
        if word in count:
            count[word] += 1
        else
        count[word] = 1

words = count.keys()
sorted_words = sorted(words, key lambda w: count[w])
for word in sorted_words:
    print "%8d %s" % (count[word], word)

File: week07/frequency.pl


while (my $line = <>) {
    $line =~ tr/A-Z/a-z/;
    foreach my $word ($line =~ /[a-z]+/g) {
        $count{$word}++;
    }
}

my @words = keys($count);
my @sorted_words = sort({$count{$a} <=> $count{$b}} @words);

